src/jobs/process-blocks.js

```diff
@@ -51,124 +66,144 @@ export default {
+            //  run the remove() since there can't possibly be any events
+            //  already indexed for this block (unless, I suppose,
+            //  minConfirmations changes to a smaller number in the config 🤔)
+            if (job.data.nextBlockNumberToProcess >= currentBlockNumberWithMinConfirmations) {
+              return currentBlockNumberWithMinConfirmations
+            }
+
+            return models.BlockchainEvent
+              .remove({ blockNumber: { $gte: job.data.nextBlockNumberToProcess } })
+              .then(({ result }) => {
+                if (result.n > 0) {
+                  logger.info(`[${this.name}]`, `removed ${result.n} existing BlockchainEvent records with blockNumber >= ${job.data.nextBlockNumberToProcess}`)
+                }
+                return currentBlockNumberWithMinConfirmations
+              })
 
-            const currentBlockNumberWithMinConfirmations = currentBlockNumber - config.blockchain.minConfirmations
+          })
+
+          .then((currentBlockNumberWithMinConfirmations) => {
 
-            logger.verbose(`[${this.name}]`, 'current block number is', currentBlockNumber)
             logger.verbose(`[${this.name}]`, `current block number with ${config.blockchain.minConfirmations} confirmations is`, currentBlockNumberWithMinConfirmations)
 
             if (job.data.nextBlockNumberToProcess > currentBlockNumberWithMinConfirmations) {
               logger.verbose(`[${this.name}]`, `no blocks to process, waiting until ${job.data.nextBlockNumberToProcess} has at least ${config.blockchain.minConfirmations} confirmations`)
               return job
             }
 
-            return this.processBlocks(job.data.nextBlockNumberToProcess, currentBlockNumberWithMinConfirmations)
-              .then((nextBlockNumberToProcess) => {
-                job.data.nextBlockNumberToProcess = nextBlockNumberToProcess
+            // only get "chunkSize" blocks at a time, to prevent the job from
+            //  trying to do too much and crashing the process when rebuilding
+            //  the database from scratch
+            const fromBlock = job.data.nextBlockNumberToProcess
+            const toBlock = ((fromBlock + config.blockchain.chunkSize) - 1 > currentBlockNumberWithMinConfirmations)
+              ? currentBlockNumberWithMinConfirmations
+              : (fromBlock + config.blockchain.chunkSize) - 1
+
+            return this.processBlocks(fromBlock, toBlock)
+              .then(() => {
+                job.data.nextBlockNumberToProcess = toBlock + 1
                 job.markModified('data')
                 return job.save()
               })
 
           })
           .catch((error) => {
-            logger.info(`[${this.name}]`, 'could not get currentBlockNumber:', error)
+            logger.warn(`[${this.name}]`, 'could not process blocks:', error)
           })
 
       })
 
   },
 
-  processBlocks(fromBlockNumber, toBlockNumber) {
```

Comment:
From here on down the diff gets a little intense, but it's because the old `processBlocks()` was removed and what used to be `processBlock()` is now `processBlocks()`. That combined with a lot of new indentation makes it look like a lot more changed here than actually did.